### PR TITLE
Update `Animation` tests

### PR DESCRIPTION
**Pull request by vanvoljg**
Here are some updated Animation tests.

There's a lot of literal fixtures, but I'm not sure how else to test that the macros are doing the right thing...

### CODE REVIEW

File: test/rgb_matrix/animation_test.exs [hunk 1]
```diff
@@ -1,12 +1,190 @@
...
+    :create_config,
+    :create_animation,
+    :create_schema
+  ]
 
   test "can get an animation type's human-readable name" do
-    defmodule MockAnimation.HueWave do
+    assert Animation.type_name(MockHueWave) == "Mock Hue Wave"
+    assert Animation.type_name(MockAnimations.Mock.HueWave) == "Hue Wave"
+    assert Animation.type_name(Mock.Animations.Hue.Wave) == "Wave"
```
Comment (amclain) [inline test/rgb_matrix/animation_test.exs:20]: ⚠️ (recommended)

I recommend putting this back to the assertion it was before, as these changes are locking in behavior we _don't_ want, or they're redundant. The purpose of the original test was to make sure things don't break with the animation name until we get to the better design of an animation providing it's own human-readable name, like with a `name "Hue Wave"` macro. The previous test should have accounted for that without breaking.

```ex
assert Animation.type_name(MockAnimation.HueWave) == "Hue Wave"
```

File: test/rgb_matrix/animation_test.exs [hunk 2]
```diff
@@ -1,12 +1,190 @@
...
+  test "can update the config of the provided animation", %{
+    animation: animation,
+    config: config
+  } do
+    params = %{"test_integer" => "5", test_option: :b}
+    updated_config = %{config | test_option: :b, test_integer: 5}
+
+    assert animation = %Animation{} = Animation.update_config(animation, params)
+    refute animation.config == config
+    assert animation.config == updated_config
```
Comment (amclain) [inline test/rgb_matrix/animation_test.exs:70]: ℹ️ (optional / question)

Doesn't the nature of `assert animation.config == updated_config` mean that if it passes the `refute` will always fail, and vice versa? In other words, only one of these tests is necessary (preferably the assertion verifying the intended state -- `assert`).

**Review by amclain — APPROVED** [review]
Just a couple comments, but overall looks good. 👍

File: test/rgb_matrix/animation_test.exs (hunk 2 above)
Comment (vanvoljg) [reply test/rgb_matrix/animation_test.exs:70]: Hmm... I see your point, yeah. I suppose an incorrectly-updated config should be caught by the Config tests, not the Animation tests.

File: test/rgb_matrix/animation_test.exs (hunk 1 above)
Comment (vanvoljg) [reply test/rgb_matrix/animation_test.exs:20]: Yeah, good point.

File: lib/rgb_matrix/animation.ex [hunk 3]
```diff
@@ -80,7 +81,7 @@ defmodule RGBMatrix.Animation do
   @doc """
   Returns the next state of an animation based on its current state.
   """
-  @spec render(animation :: t) :: {render_in, [RGBMatrix.any_color_model()], t}
+  @spec render(animation :: t) :: {render_in, frame, t}
```
[the remark this answers is not in the document]
Comment (vanvoljg) [reply lib/rgb_matrix/animation.ex:84]: It was a little trick meant to keep the extraneous parens out of the `@spec`. No good?

[the remark this answers is not in the document]
Comment (doughsay) [reply lib/rgb_matrix/animation.ex:84]: It's gonna confuse the ExDoc.  It'll look like two different types, which isn't ideal.  We're just gonna have to suck up the parens...

**Review by doughsay — DISMISSED** [review]

File: lib/rgb_matrix/animation.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (vanvoljg) [reply lib/rgb_matrix/animation.ex:84]: Bummer. I really wish remote types... eh, whatever. Fix incoming...

[the remark this answers is not in the document]
Comment (amclain) [reply lib/rgb_matrix/animation.ex]: >It was a little trick meant to keep the extraneous parens out of the `@spec`. No good?

I'd prioritize having the code in the right namespaces over perfecting the syntax.

**Review by amclain — APPROVED** [review]

**Pull request merged**